tensorflow_addons/metrics/matthews_correlation_coefficient_test.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Matthews Correlation Coefficient Test."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+from tensorflow.keras import layers
+import numpy as np
+from tensorflow_addons.utils import test_utils
+from tensorflow_addons.metrics import MatthewsCorrelationCoefficient
+
+
+@test_utils.run_all_in_graph_and_eager_modes
+class MatthewsCorrelationCoefficientTest(tf.test.TestCase):
+    def test_config(self):
+        # mcc object
+        mcc1 = MatthewsCorrelationCoefficient(num_classes=1)
+        self.assertEqual(mcc1.num_classes, 1)
+        self.assertEqual(mcc1.dtype, tf.float32)
+        # check configure
+        mcc2 = MatthewsCorrelationCoefficient.from_config(mcc1.get_config())
+        self.assertEqual(mcc2.num_classes, 1)
+        self.assertEqual(mcc2.dtype, tf.float32)
+
+    def initialize_vars(self, n_classes, input_dtype):
+        mcc = MatthewsCorrelationCoefficient(
+            num_classes=n_classes, dtype=input_dtype)
+        self.evaluate(tf.compat.v1.variables_initializer(mcc.variables))
+        return mcc
+
+    def update_obj_states(self, obj, gt_label, preds):
+        update_op = obj.update_state(gt_label, preds)
+        self.evaluate(update_op)
+
+    def check_results(self, obj, value):
+        self.assertAllClose(value, self.evaluate(obj.result()), atol=1e-5)
+
+    def test_binary_classes(self):
+        gt_label = tf.constant([[1.0], [1.0], [1.0], [0.0]], dtype=tf.float32)
+        preds = tf.constant([[1.0], [0.0], [1.0], [1.0]], dtype=tf.float32)
+        # Initialize
+        mcc = self.initialize_vars(n_classes=1, input_dtype=tf.float32)
```

Comment:
Umm, shouldn't this be `n_classes=2`? I mean `binary_classes` should have two classes, say 0 and 1, right? Please correct me if I'm wrong.

Also, the semantic meaning of `dtype` in `tf.keras.metrics.Metric` should be the output dtype instead of input dtype.
https://github.com/tensorflow/tensorflow/blob/r2.0/tensorflow/python/keras/metrics.py#L451

Reply:
> Umm, shouldn't this be `n_classes=2`? I mean `binary_classes` should have two classes, say 0 and 1, right? Please correct me if I'm wrong.

Thanks @WindQAQ. The number of class should be one, if multiple classes will have something like this, `[[1.0, 1.0], [1.0, 1.0]]`, which num of class is 2.

Reply:
> Also, the semantic meaning of `dtype` in `tf.keras.metrics.Metric` should be the output dtype instead of input dtype.
> https://github.com/tensorflow/tensorflow/blob/r2.0/tensorflow/python/keras/metrics.py#L451

Thanks for pointing this out @WindQAQ. Since it's `optional data type of the metric result`, I'll remove the data type. What do you think?